lib/nrf_modem_lib/custom_trace_medium/trace_medium_printk.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/zephyr.h>
+#include <modem/trace_medium.h>
+
+TRACE_MEDIUM_REGISTER(printk, trace_medium_printk_init, trace_medium_printk_deinit, trace_medium_printk_write);
+
+#define NUM_TRACES_BEFORE_PRINT 500
+static uint32_t count;
+static uint32_t total_count = 0;
+
+static int trace_medium_printk_init(void)
+{
+	count = NUM_TRACES_BEFORE_PRINT;
+	printk("trace medium printk initialized\n");
+	return 0;
+}
+
+static int trace_medium_printk_deinit(void)
+{
+	printk("trace medium printk deinitialized\n");
+	return 0;
+}
+
+static void trace_medium_printk_write(const uint8_t *data, uint32_t len)
+{
+	ARG_UNUSED(data);
+	ARG_UNUSED(len);
+
+	if (count == 0) {
+		printk("Trace data received %d times\n", total_count);
+		count = NUM_TRACES_BEFORE_PRINT;
+		total_count += NUM_TRACES_BEFORE_PRINT;
+	} else {
+		count--;
```

Comment:
Instead of counting the number of received traces to decide if we should print we could print how much is received every x seconds.